web/packages/teleterm/src/ui/ModalsHost/modals/ReAuthenticate/ReAuthenticate.tsx

```diff
@@ -80,45 +80,54 @@ export const ReAuthenticate: FC<{
   // TODO(ravicious): Use a profile name here from the URI and remove the dependency on
   // clustersService. https://github.com/gravitational/teleport/issues/33733
   const rootClusterUri = routing.ensureRootClusterUri(clusterUri);
+  const rootCluster = clustersService.findRootClusterByResource(rootClusterUri);
   const rootClusterName =
-    clustersService.findRootClusterByResource(rootClusterUri)?.name ||
-    routing.parseClusterName(rootClusterUri);
+    rootCluster?.name || routing.parseClusterName(rootClusterUri);
+  const rootClusterProxyHost =
+    // As a fallback, we read the proxy hostname from the URI. One small issue is that URIs don't
+    // include the port number, so if the actual proxy host has a port number other than 443,
+    // rootClusterProxyHost will not point to the proxy service.
+    // In practice though we should not end up in a situation where this modal is shown but the
+    // cluster does not exist in the app.
+    rootCluster?.proxyHost || routing.parseClusterName(rootClusterUri);
   const clusterName =
     clustersService.findCluster(clusterUri)?.name ||
     routing.parseClusterName(clusterUri);
   const isLeafCluster = routing.isLeafCluster(clusterUri);
 
-  const totpPrompt = useMemo(() => {
```

Comment:
`useMemo` wasn't strictly necessary here, it was not a costly operation. Though defining `totpPrompt` within a function looked indeed slightly better, without it we either have to use `let` like I did or `const $totpPrompt` with a ternary.

I renamed `totpPrompt` to `$totpPrompt`, as this is a convention that we use for when a variable holds JSX objects.

Reply:
Good to know about the `$` convention :+1: